### ISSUE TITLE
Throw helpful error if `privateKey` is set to only the first line, e.g. `-----BEGIN RSA PRIVATE KEY-----`

### ISSUE THREAD

**Issue by gr2m** [gr2m/universal-github-app-jwt#71]
Follow up to https://github.com/octokit/auth-app.js/issues/465#issuecomment-1564998327, reported by @reza-ebrahimi

The first line might be `-----BEGIN PRIVATE KEY-----` or `-----BEGIN RSA PRIVATE KEY-----`, maybe we will support other variations in future. But they all have the format `-----BEGIN * KEY-----`

I would suggest we add a check like 

```js
if (/-----BEGIN * KEY-----/.test(privateKey?.trim()) {
  throw new Error("privateKey only contains the first line. Try replacing line breaks with \n")
}
```

**Comment by gr2m** [comment]
> maybe we will support other variations in future

in particular I would love to see if we can support `-----BEGIN OPENSSH PRIVATE KEY-----`, which is the format that 1Password stores ssh keys in its latest version

**Comment by reza-ebrahimi** [comment]
Is there any `schema` or `standard` for private key content? Or probably an open source `schema validator` javascript / typescript library?

**Comment by reza-ebrahimi** [comment]
Not sure if directly related to PK validation, but I found these:

- [The 'jsrsasign' (RSA-Sign JavaScript Library) is an opensource free cryptography library supporting RSA/RSAPSS/ECDSA/DSA signing/validation, ASN.1, PKCS#1/5/8 private/public key, X.509 certificate, CRL, OCSP, CMS SignedData, TimeStamp, CAdES and JSON Web Signature/Token in pure JavaScript.](https://github.com/kjur/jsrsasign)
- [RSASSA-PSS signature validation failure](https://github.com/digitalbazaar/forge/issues/419)
- [CSR validation fail with own private key](https://github.com/digitalbazaar/forge/issues/684)
- [How to validate PGP public key written by a user?](https://stackoverflow.com/a/57679988/1789729)

**Comment by reza-ebrahimi** [comment]
A simpler approach with `regex`. The expression below will match strings ([Regex test link](https://regex101.com/r/tohNai/1)):

```
^-----BEGIN.*PRIVATE KEY-----$
```  
- starts with `-----BEGIN`
- end with `PRIVATE KEY-----`

**Comment by gr2m** [comment]
yeah good idea with the regex. I'd just add two spaces around `.*`

```js
/^-----BEGIN .* PRIVATE KEY-----$/
```

**Comment by gr2m** [comment]
> Is there any `schema` or `standard` for private key content

We don't need a sophisticated private key validation library. This is about the very specific and common problem that only the first line is passed as the `privateKey` value

**Comment by reza-ebrahimi** [comment]
Created a PR: https://github.com/octokit/auth-app.js/pull/482
